src/database-sqlite3.h

```diff
@@ -30,37 +32,99 @@ extern "C" {
 class Database_SQLite3 : public Database
 {
 public:
-	Database_SQLite3(const std::string &savedir);
-	~Database_SQLite3();
+	virtual ~Database_SQLite3();
 
 	void beginSave();
 	void endSave();
 
-	bool saveBlock(const v3s16 &pos, const std::string &data);
-	void loadBlock(const v3s16 &pos, std::string *block);
-	bool deleteBlock(const v3s16 &pos);
-	void listAllLoadableBlocks(std::vector<v3s16> &dst);
 	bool initialized() const { return m_initialized; }
+protected:
+	Database_SQLite3(const std::string &savedir, const std::string &dbname);
 
-private:
-	// Open the database
-	void openDatabase();
-	// Create the database structure
-	void createDatabase();
 	// Open and initialize the database if needed
 	void verifyDatabase();
 
-	void bindPos(sqlite3_stmt *stmt, const v3s16 &pos, int index = 1);
+	// Convertors
+	inline void str_to_sqlite(sqlite3_stmt *s, const int iCol, const std::string &str) const
+	{
+		sqlite3_vrfy(sqlite3_bind_text(s, iCol, str.c_str(), str.size(), NULL));
+	}
+
+	inline void str_to_sqlite(sqlite3_stmt *s, const int iCol, const char *str) const
+	{
+		sqlite3_vrfy(sqlite3_bind_text(s, iCol, str, strlen(str), NULL));
```

Comment:
i reverted the code factorization as our player name model doesn't work when stored into a std::string, sqlite receive an empty string